handlers/handlers.go

```diff
@@ -478,26 +469,30 @@ func readPreviousReleases(w http.ResponseWriter, req *http.Request, cfg *config.
 	}
 
 	basePage := rend.NewBasePageModel()
-
 	err = validateCurrentPage(ctx, cfg, validatedQueryParams, searchResp.Count)
 	if err != nil {
 		validationErrs = append(validationErrs, core.ErrorItem{
 			Description: core.Localisation{
 				Text: "current page exceeds total pages",
 			},
 		})
-		m := mapper.CreatePreviousReleasesPage(cfg, req, basePage, validatedQueryParams, &searchModels.SearchResponse{}, lang, zebedeeCli.HomepageContent{}, "", navigationCache, template, cache.Topic{}, validationErrs, pageData)
+		m, _ := mapper.CreatePreviousReleasesPage(cfg, req, basePage, validatedQueryParams, &searchModels.SearchResponse{}, lang, zebedeeCli.HomepageContent{}, "", navigationCache, template, cache.Topic{}, validationErrs, pageData)
 		rend.BuildPage(w, m, template)
 		return
 	}
 
-	m := mapper.CreatePreviousReleasesPage(cfg, req, basePage, validatedQueryParams, searchResp, lang, homepageResp, "", navigationCache, template, cache.Topic{}, validationErrs, pageData)
+	m, err := mapper.CreatePreviousReleasesPage(cfg, req, basePage, validatedQueryParams, searchResp, lang, homepageResp, "", navigationCache, template, cache.Topic{}, validationErrs, pageData)
+	if err != nil {
+		log.Error(ctx, "cannot render previous releases page", err)
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
```

Comment:
I don't think this is the right pattern to be following in this instance. We should be rendering with 0 results, not throwing an error.

Reply:
👍